apps/meteor/app/livechat/server/business-hour/BusinessHourManager.ts

```diff
@@ -115,12 +159,27 @@ export class BusinessHourManager {
 			callbacks.priority.HIGH,
 			'business-hour-livechat-on-save-agent-department',
 		);
+		callbacks.add(
+			'livechat.afterDepartmentDisabled',
+			this.behavior.onDepartmentDisabled.bind(this),
+			callbacks.priority.HIGH,
+			'business-hour-livechat-on-department-disabled',
+		);
+		// TODO: Binding the correct "this" for the context of onDepartmentArchived. Should we do the same for other callbacks?
```

Comment:
this comment is not really necessary. If you have a question, or believe something is wrong, please find out and act accordingly. this kind of thought scattered in the code does not add anything

btw: passing a method of an object as a callback and expecting this to remain the same is wrong and amateur, whoever does this should be aware
```suggestion
```

Reply:
The comment is indeed outdated, so thanks for removing it.

As you see, the idea of the calls never implies they're "the same", rather, than for some reason we're passing `this` which in the context is an instance of `businessHourManager` instead of `this.behavior` since the calls are coming from that context. However, none of the calls actually reference `this` inside, so the comment was meant to handle it later and test accordingly, since for now, it's not causing any harm 🤗